From bc8f09d8dc2ff5caf1fcfc6b6891909950c13ef5 Mon Sep 17 00:00:00 2001
From: Prakash Brahmajyosyula <bprakash@marvell.com>
Date: Tue, 16 Oct 2018 09:59:19 +0300
Subject: [PATCH 0262/1051] net: octeontx2: Update Nix rqpool cnt when rx
 queues changed.

This patch fixes translation faults when rx/tx queues changed.

Signed-off-by: Prakash Brahmajyosyula <bprakash@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c
index abe3d99b10a8..387ff7539c16 100644
--- a/drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/otx2_ethtool.c
@@ -238,6 +238,7 @@ static int otx2_set_channels(struct net_device *dev,
 
 	pfvf->hw.rx_queues = channel->rx_count;
 	pfvf->hw.tx_queues = channel->tx_count;
+	pfvf->hw.rqpool_cnt = channel->rx_count;
 	err = otx2_set_real_num_queues(dev, pfvf->hw.tx_queues,
 				       pfvf->hw.rx_queues);
 	pfvf->qset.cq_cnt = pfvf->hw.tx_queues +  pfvf->hw.rx_queues;
-- 
2.17.1

